### PR TITLE
Fix nonce calculation for geth

### PR DESCRIPTION
**Pull request by palango**
Before I got a warning that `str`  and `int` could not be concatenated.

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/paul/Projects/brainbot/raiden/tools/scenario-player/scenario_player/__main__.py", line 9, in <module>
    prog_name='python -m scenario_player',
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/paul/Projects/brainbot/raiden/venv/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/paul/Projects/brainbot/raiden/tools/scenario-player/scenario_player/main.py", line 129, in main
    runner = ScenarioRunner(account, chain_rpc_urls, auth, Path(data_path), scenario_file)
  File "/Users/paul/Projects/brainbot/raiden/tools/scenario-player/scenario_player/runner.py", line 96, in __init__
    gas_price_strategy=self.scenario.gas_price_strategy,
  File "/Users/paul/Projects/brainbot/raiden/raiden/network/rpc/client.py", line 440, in __init__
    address_checksumed,
  File "/Users/paul/Projects/brainbot/raiden/raiden/network/rpc/client.py", line 176, in geth_discover_next_available_nonce
    return max(pending.keys()) + 1
TypeError: can only concatenate str (not "int") to str
```

### CODE REVIEW

**Comment by LefterisJP** [comment]
@palango That's rather strange. What geth version are you using? We should have seen this break in our tests.

**Comment by palango** [comment]
```
VERSION:
   1.8.22-stable
```

**Comment by LefterisJP** [comment]
@palango Same as in our tests.

So I am kind of confused why we don't see this in the tests. Can you wait till I try to reproduce it?

**Comment by palango** [comment]
Sure.

**Comment by LefterisJP** [comment]
@palango Okay reproduced in a test. This never worked ffs. Will push in your PR. Thank you so much for finding this my friend!

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -167,13 +167,14 @@ def geth_discover_next_available_nonce(
     # the user, these will be the younger transactions. Because this needs the
     # largest nonce, queued is checked first.
 
+    address = to_checksum_address(address)
```
Comment (LefterisJP) [inline raiden/network/rpc/client.py]: The absence of this is why we did not see this in real life and why you only saw this from the SP. And this was yet another bug.

Web3 returns an `AttributeDict` that looks like this:

```
KeysView(AttributeDict({'10':
'0x50607A6053295b4d7E3A4F38242c5C0a3e47793a: 0 wei + 50000 gas
1000000000 wei', '100': '0x2C3154203f5127552664763D3053343d2f756151: 0
wei + 50000 gas 1000000000 wei',
```

And its addresses are not only hex, but they are also checksummed. As
such we need to also use checksummed addressed in this query or else
we will never get a correct response from geth.

**Comment by palango** [comment]
Changes looks good, added one question, but that should be reasonable fast on a private chain anyways.

**Pull request merged**